### PR TITLE
WIP: Logging for Login/Auth process

### PR DESCRIPTION
**Pull request by jrose-stac**
Add info/error level logging statements to different pieces of login/2FA auth process.

### CODE REVIEW

File: server/models/users.js [hunk 1]
```diff
@@ -347,15 +357,20 @@ module.exports = class User extends Model {
       }
     }
     console.info(redirect)
+    WIKI.logger.info(`users.afterLoginChecks: Group lookup and redirect completed`)
+    WIKI.logger.info(redirect)
```
Comment (bxjw) [inline server/models/users.js:361]: is this different from the console.info(redirect) two lines above

File: server/models/users.js [hunk 2]
```diff
@@ -467,23 +488,31 @@ module.exports = class User extends Model {
...
       })
       if (user) {
+        WIKI.logger.info(`users.loginTFA: Valid user found for login. User ID ${user.id}`)
         if (user.verifyTFA(securityCode)) {
+          WIKI.logger.info(`users.loginTFA: TFA Code Valid for User ID ${user.id}`)
           if (setup) {
+            WIKI.logger.info(`users.loginTFA: Enabling TFA for User ID ${user.id}`)
             await user.enableTFA()
           }
+          WIKI.logger.info(`users.loginTFA: Finish Two FActor Authentication Login.`)
```
Comment (bxjw) [inline server/models/users.js]: total nitpick: correct the capitalization

File: server/models/users.js [hunk 3]
```diff
@@ -467,23 +488,31 @@ module.exports = class User extends Model {
...
           return WIKI.models.users.afterLoginChecks(user, context, { skipTFA: true })
         } else {
+          WIKI.logger.error(`users.loginTFA: Verify FTA failed for User ID ${user.id}.`)
```
Comment (bxjw) [inline server/models/users.js]: nitpick: TFA not FTA

File: server/models/users.js [hunk 4]
```diff
@@ -467,23 +488,31 @@ module.exports = class User extends Model {
...
           throw new WIKI.Error.AuthTFAFailed()
         }
       }
     }
+    WIKI.logger.error(`users.loginTFA: Verify FTA failed for User ID ${user.id}.`)
```
Comment (bxjw) [inline server/models/users.js]: nitpick: TFA not FTA

**Review by bxjw — CHANGES_REQUESTED** [review]
looks fine to me - where does the logger go to in terms of log group, do you know?

File: server/models/userKeys.js [hunk 5]
```diff
@@ -62,10 +65,12 @@ module.exports = class UserKey extends Model {
...
       if (DateTime.utc() > DateTime.fromISO(res.validUntil)) {
+        WIKI.logger.error(`userKeys.WIKI: User auth token expired for User with ID ${res.id}.`)
         throw new WIKI.Error.AuthValidationTokenInvalid()
       }
       return res.user
     } else {
+      WIKI.logger.error(`userKeys.WIKI: User auth token not found. Token: ${token}`)
       throw new WIKI.Error.AuthValidationTokenInvalid()
     }
   }
```
Comment (ffaristocrat) [inline server/models/userKeys.js:76]: Might as well put a log statement in destroyToken as well. Issue could be that it's getting created and trashed immediately for some reason.

**Review by ffaristocrat — COMMENTED** [review]
LGTM

File: server/models/users.js (hunk 1 above)
Comment (jrose-stac) [reply server/models/users.js:361]: console.info doesn't actually output anything unless you're running locally

**Pull request closed without merging**